trace.py

```diff
@@ -195,7 +195,31 @@ def define_val_type(val): # recursive function used to change strings into typed
 # Load binary passed in argv[1] and check we only have one arg
 if len(sys.argv) < 2:
     print("Please pass in at least one argument (executable you wish to run)!")
+    os.remove("output.txt")
     exit()
+# Test whether this file exists by checking if the path exists
+if not os.path.exists(sys.argv[1]):
+    print("This executable does not exist!")
+    os.remove("output.txt")
+    exit()
+# Test whether the file is an executable
+if not os.access(sys.argv[1], os.X_OK):
+    print("This file is not an executable!")
+    os.remove("output.txt")
+    exit()
+
+# Testing time it took program to run
+os.popen(f'(time {sys.argv[1]}) &> time.txt')
+time.sleep(5)
+run_time = open('time.txt')
+if not list(run_time.read().splitlines()):
+    print(list(run_time.read().splitlines()))
+    print("There may be an infinite loop in the code")
+    os.remove("output.txt")
+    os.remove("time.txt")
+    run_time.close()
```

Comment:
What happens to the process that you started with os.popen? is it still running in the background after trace.py exits? If this is how it behaves, I would recommend explicitly killing the process before exiting

Reply:
I will look into that!